distributed/deploy/ssh.py

```diff
@@ -311,11 +311,20 @@ def SSHCluster(
         kwargs.setdefault("worker_addrs", hosts)
         return OldSSHCluster(**kwargs)
 
+    if isinstance(connect_options, list):
+        if len(connect_options) != len(hosts):
```

Comment:
This may be slightly more readable as a single `if`-statement

```python
if isinstance(connect_options, list) and len(connect_options) != len(hosts):
```